File: src/com/lulu/leetcode/链表/java/ListNode.java
package com.lulu.leetcode.链表.java;

public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        val = x;
    }


    public static ListNode createLinkedList(int[] arr) {
        if (arr.length == 0) {
            return null;
        }
        ListNode head = new ListNode(arr[0]);
        ListNode curNode = head;
        for (int i = 0; i < arr.length; i++) {
            curNode.next = new ListNode(arr[i]);
            curNode = curNode.next;
        }
        return head;
    }

    public static void printLinedList(ListNode head) {
        if (head == null) {
            System.out.println("null");
            return;
        }

        ListNode curNode = head;
        while (curNode != null) {
            System.out.print(curNode.val + "->");
            curNode = curNode.next;
        }
        System.out.println("null");
    }

    public static void main(String[] args) {
        ListNode listNode = ListNode.createLinkedList(new int[]{1, 2, 3, 4, 5, 6});
        ListNode.printLinedList(listNode);
    }
}

File: src/com/lulu/leetcode/链表/ListNode.kt
package com.lulu.leetcode.链表

class ListNode(var `val`: Int) {
    var next: ListNode? = null

    companion object {
        public fun createLinkedList(arr: IntArray, n: Int): ListNode? {
            if (n == 0) {
                return null
            }
            val head: ListNode? = ListNode(arr[0])
            var curNode = head
            for (i in 1 until n) {
                curNode?.next = ListNode(arr[i])
                curNode = curNode?.next
            }
            return head
        }

        public fun printLinedList(head: ListNode?) {
            if (head == null) {
                println("null")
                return
            }

            var curNode: ListNode? = head
            while (curNode != null) {
                print("${curNode.`val`}->")
                curNode = curNode.next
            }
            println("null")
        }

        @JvmStatic
        fun main(args: Array<String>) {
            val head = createLinkedList(intArrayOf(1, 2, 3, 4, 5, 6), 6)
            printLinedList(head)
        }
    }


}